imap_processing/ialirt/process_ephemeris.py

```diff
@@ -158,6 +135,66 @@ def calculate_azimuth_and_elevation(
     return np.asarray(azimuth), np.asarray(elevation)
 
 
+@typing.no_type_check
+def calculate_doppler(
+    longitude: float,
+    latitude: float,
+    altitude: float,
+    observation_time: Union[float, np.ndarray],
+    target: str = SpiceBody.IMAP.name,
+) -> Union[float, ndarray[float]]:
+    """
+    Calculate the doppler velocity.
+
+    Notes about the spkezr function:
+    The function returns the state of the target (state) and the light time.
+    The first three components of state represent the x-, y- and z-components of the
+    target's position; the last three components form the corresponding velocity vector.
+
+    Parameters
+    ----------
+        longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian,
+        negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative
+        to south.
+    altitude : float
+        Altitude in kilometers.
+    observation_time : float or np.ndarray
+        Time at which the state of the target relative to the observer
+        is to be computed. Expressed as ephemeris time, seconds past J2000 TDB.
+    target : str (Optional)
+        The target body. Default is "IMAP".
+
+    Returns
+    -------
+    doppler : float or np.ndarray[float]
+        Doppler velocity in kilometers per second.
+    """
+    ground_station_position_ecef = latitude_longitude_to_ecef(
+        longitude, latitude, altitude
+    )
+
+    if not isinstance(observation_time, np.ndarray):
+        observation_time = [observation_time]
+
+    # TODO: consider improving vectorizing this for loop by moving everything besides
+    #  the spkezr outside of the for loop,
+    # TODO: which would require the axis argument
+    # TODO: error handling with a try/except block
+    doppler = []
+
+    for timestamp in observation_time:
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.spkezr
+        state, light_time = spice.spkezr(target, timestamp, "ITRF93", "LT+S", "EARTH")
+        # re-orient spacecraft position relative to the ground station
+        state = state - np.pad(ground_station_position_ecef, (0, 3), "constant")
```

Comment:
The use of np.pad is unintuitive to me. Could this just be:
```suggestion
        state[0:3] -= ground_station_position_ecef
```

Reply:
If I remove the use of np.pad, I'll have to figure out a way to also do this position correction for a state of multiple rows. I haven't figured out how to condense this to one line, so it would be separated into some if/else logic. As it is, I can address both cases (one row vs. multiple rows) in one line.